# RI 53 Radio Network - UMTS

# Exercise 1:

a. Find the Hadamard matrix H16.

b. Find SF (spreading factor) for a W-CDMA network with 3.84 Mcps channel bit rate and 240 kbps information bandwidth (user information).

# Solution

a. Find the Hadamard matrix H16.


In [5]:
from IPython.display import Image
Image(url= "matrix H16.png", width=300, height=300)

In [3]:
# draw th Hadamard matrix H16
H16 = np.array([[1, 1, 1, 1],
                [1, -1, -1, -1],
                [1, -1, 1, -1],
                [1, 1, -1, 1]])

# print the Hadamard matrix H16
print(H16)

[[ 1  1  1  1]
 [ 1 -1 -1 -1]
 [ 1 -1  1 -1]
 [ 1  1 -1  1]]


b. Find SF (spreading factor) for a W-CDMA network with 3.84 Mcps channel bit rate and 240 kbps information bandwidth (user information).

SF = Chip rate/ Bit Rate = 3.84* 10^6 /240* 10^3 =16 = 2^4 = 4 level

In [8]:
# find the spreading factor
import math
bitRate = 3.84 # Mcps
informationBandwidth = 240 # kbps

SF = bitRate * pow(10, 6) / (informationBandwidth * pow(10, 3))
print("spreading factor", math.log(SF,2))

spreading factor 4.0


# Exercise 2:

DS-CDMA system has a spreading code rate of 3.84 Mchips. User data rate is 15 kbits.

How many users can one cell serve when the required level of performance requires an SIR 9 dB?

# Solution
    The capacity in CDMA system is interference limited. The interference per bit is described
    as energy per bit divided by interference in the transmission bandwidth.

    In CDMA the interference in communication bandwidth is generated by all other users as
    a zero-mean AWGN process.


In [10]:
Image(url= "UMTS2.png", width=800, height=200)

In [11]:
Image(url= "UMTS3.png", width=600, height=400)

In [14]:
spreadingCoderate = 3.84 # Mchips
dataRate = 15 #kbits
SIR = 9


numberOfUsers = spreadingCoderate * pow(10, 6) / (SIR * dataRate * pow(10, 3)) + 1
print("number of users", numberOfUsers)

number of users 29.444444444444443


# Exercise 3:

How many orthogonal codes are available for the voice users with coded data rate 30 kbits in CDMA uplink, if there are allocated codes for 6 data users with user data rates [240 240 240 480 480 480]. The spread chip rate is 3.84 Mchips.

In [15]:
Image(url= "umts-ex-3.png", width=600, height=400)

# Solution 3
As we know in CDMA uplink each user has its own orthogonal Walsh code. The codes are selected from the channelization code tree. For the 3.84Mchips and 30 kbps data users there are total 128 (2^7 ) different code words => the code tree is 7 levels deep.

In [16]:
Image(url= "umts-ex3-s.png", width=700, height=300)

In [20]:
userDataRates = [240, 240, 240, 480, 480, 480]
coddedDataRate = 30 # kbits
spreadChipRate = 3.84 # Mchips

SF240 = math.log(spreadChipRate * pow(10, 6) / (240 * pow(10, 3)),2)
print("spreading factor SF240", SF240)

SF480 = math.log(spreadChipRate * pow(10, 6) / (480 * pow(10, 3)),2)
print("spreading factor SF480", SF480)

spreading factor SF240 4.0
spreading factor SF480 3.0


# Exercise 4:
In addition to using a code, the chips are placed on a frequency/time slot. Each frequency is given a channel number: UARFCN (UTRAN Absolute Radio Frequency Channel Number).

We use 5 MHz channels bandwidth where the carrier is in the center. We associate this UARFCN number and the frequency by the following equations depending on the band:

▪ In the 2 GHz band I FDD: 1920-1980 MHz uplink and 2110-2170 MHz downlink
        UARFCN = 5 x downlink carrier freq in MHz 
  UARFCN numbers from 10562 to 10838,

▪ In the 900 MHz band replacing GSM and E-GSM (880 to 915 MHz uplink and 925 to 960
MHz downlink = band VIII)
        UARFCN = 5 x (downlink carrier freq in MHz - 340).
UARFCN Numbers from 2937 to 3088,

1) What are the uplink and downlink frequencies for UARFCN 10787?

In [22]:
Image(url= "umts-ex-4-1-s.png", width=600, height=120)

In [30]:
UARFCN = 10787
F_downLink = UARFCN / 5
print("F_downLink", F_downLink)
F_upLink = F_downLink - 190
print("F_upLink", F_upLink)

F_downLink 2157.4
F_upLink 1967.4


2) The UMTS communication analyzer indicates communication with UARFCN = 3037. Which operator is carrying the communication based on the channel assignment given for service providers who have acquired operating licenses?

In [23]:
Image(url= "umts-ex-4b.png", width=1200, height=200)

In [24]:
Image(url= "umts-ex-4b-1.png", width=1200, height=200)

In [26]:
Image(url= "umts-ex-4-b-s.png", width=600, height=120)

In [27]:
UARFCN = 3037
F_DL = UARFCN / 5 + 340
print("F_DL", F_DL)

F_DL 947.4


3) What is the network used by the phone at the time of the analysis (operator and type of network)?

Note that the MCC for metropolitan France is 208, and the MNC for the operator is 01 = Orange 10 = SFR 20 = BouyguesTelecom 15 = Freemobile)

In [29]:
Image(url= "umts-ex-4-c.png", width=400, height=400)

Answer: WCDMA900 at band 8, operator is 15 = Free

4) From the screenshot of GSM&UMTS measurement software, what is the information we can find about making this call?

In [28]:
Image(url= "umts-ex-4-d.png", width=600, height=400)

Answer: Network= UMTS2100 (FDD); Operator= Orange France; Spreading factor=128; NodeB
scrambling codes 345,340; Used frequencies= 1967.4𝑀𝐻𝑧, 2157.4𝑀𝐻𝑧